src/RuleMatcher.cc

```diff
@@ -865,11 +875,10 @@ void RuleMatcher::Match(RuleEndpointState* state, Rule::PatternType type, const
                 continue;
 
             // Remember that all patterns have matched.
-            if ( ! state->matched_by_patterns.is_member(r) ) {
-                state->matched_by_patterns.push_back(r);
-                String* s = new String(data, data_len, false);
-                state->matched_text.push_back(s);
-                state->matched_text_end_of_match.push_back(match_end_pos - pre_match_pos);
+            const auto* pattern_match = state->FindRulePatternMatch(r);
+            if ( pattern_match == nullptr ) {
```

Comment:
Style nit, but I think we generally skip explicit comparisons to `nullptr`:

```suggestion
            if ( ! pattern_match ) {
```

Reply:
Actually, don't even need the tmp:

```
if ( ! state->FindRulePatternMatch(r) ) {
```